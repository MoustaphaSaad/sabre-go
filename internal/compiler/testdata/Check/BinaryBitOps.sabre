package main

func correct() {
	1 | 2
	1 & 2
	1 ^ 2
	1 &^ 2
}

func incorrect() {
	1 | 2.5
	1 & 2.5
	1 ^ 2.5
	1 &^ 2.5

	1 | true
	1 & true
	1 ^ true
	1 &^ true
}
