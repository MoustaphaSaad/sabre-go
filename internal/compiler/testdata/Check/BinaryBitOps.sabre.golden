>> 		1 | 2.5
>> 		^^^^^^^ 
Error[internal/compiler/testdata/Check/BinaryBitOp.sabre:11:2]: type mismatch in binary expression, lhs is 'int' and rhs is 'float32'
>> 		1 & 2.5
>> 		^^^^^^^ 
Error[internal/compiler/testdata/Check/BinaryBitOp.sabre:12:2]: type mismatch in binary expression, lhs is 'int' and rhs is 'float32'
>> 		1 ^ 2.5
>> 		^^^^^^^ 
Error[internal/compiler/testdata/Check/BinaryBitOp.sabre:13:2]: type mismatch in binary expression, lhs is 'int' and rhs is 'float32'
>> 		1 &^ 2.5
>> 		^^^^^^^^ 
Error[internal/compiler/testdata/Check/BinaryBitOp.sabre:14:2]: type mismatch in binary expression, lhs is 'int' and rhs is 'float32'
>> 		1 | true
>> 		^^^^^^^^ 
Error[internal/compiler/testdata/Check/BinaryBitOp.sabre:16:2]: type mismatch in binary expression, lhs is 'int' and rhs is 'bool'
>> 		1 & true
>> 		^^^^^^^^ 
Error[internal/compiler/testdata/Check/BinaryBitOp.sabre:17:2]: type mismatch in binary expression, lhs is 'int' and rhs is 'bool'
>> 		1 ^ true
>> 		^^^^^^^^ 
Error[internal/compiler/testdata/Check/BinaryBitOp.sabre:18:2]: type mismatch in binary expression, lhs is 'int' and rhs is 'bool'
>> 		1 &^ true
>> 		^^^^^^^^^ 
Error[internal/compiler/testdata/Check/BinaryBitOp.sabre:19:2]: type mismatch in binary expression, lhs is 'int' and rhs is 'bool'
